nav2_costmap_2d/src/layer.cpp

```diff
@@ -66,4 +66,33 @@ Layer::getFootprint() const
   return layered_costmap_->getFootprint();
 }
 
+void
+Layer::declareParameter(
+  const std::string & param_name, const rclcpp::ParameterValue & value)
+{
+  local_params_.insert(param_name);
+  node_->declare_parameter(getRegName(param_name), value);
+}
+
+bool
+Layer::hasParameter(const std::string & param_name)
+{
+  return node_->has_parameter(getRegName(param_name));
+}
+
+void
+Layer::undeclareAllParameters()
+{
+  std::for_each(begin(local_params_), end(local_params_), [this](const std::string & param_name) {
+     node_->undeclare_parameter(getRegName(param_name));
+  });
+  local_params_.clear();
+}
+
+std::string
+Layer::getRegName(const std::string & param_name)
```

Comment:
I'm not sure what "Reg" is in `getRegName`. Can we either rename to be more clear or expand `Reg` to its full word? I think it might mean `Regular` or `Register` but if so, I'm not sure what that's supposed to mean in this context of just applying a namespace.

Reply:
Good idea, please see new proposal